model/commitqueue/github_merge_generic.go

```diff
@@ -57,7 +57,20 @@ func (s *GithubMergePR) Initialize(env evergreen.Environment) error {
 	return nil
 }
 
-func (s *GithubMergePR) Send() error {
+func (s *GithubMergePR) Send() (err error) {
+	defer func() {
```

Comment:
I don't think this is a good use of defer functions. These are usually to close files/catch panics. If you want this to be more modular, I recommend a private function like "handle error" that you call where relevant.

Reply:
At the risk of getting dragged into a philosophical debate... :man_with_turban:
The use case matches the semantics of defer (every exit point needs to do the same thing). Although releasing resources and resolving panics are typical use cases, limiting ourselves to only the common use cases is... limiting.